### DISCUSSION TITLE
Implementing WHATWG URL

### DISCUSSION THREAD

**Discussion by Ethan-Arrowood** [jstime/jstime#147] [Ideas]
I put a little bit of time into #127 using JS to implement the URL standard.  Truthfully, I don't think I'll be able to build a better URL implementation than other, existing JS ones. 

Instead, i'd like to propose either (a) vendoring an existing implementation (maybe we pick the fastest one?), or (b) implementing it in Rust instead.

What do folks think?

**Comment by bengl** [comment]
I'd lean toward vendoring an existing implementation, since it's much easier than trying to implement it in Rust, and maybe faster. For [osgood](https://github.com/IntrinsicLabs/osgood) we used https://github.com/jsdom/whatwg-url directly. It can be swapped out for a better/faster implementation later on if one exists.

**Reply by Ethan-Arrowood** [reply]
I think jsdom/whatwg-url is a good one to use too.

**Reply by MylesBorins** [reply]
so this was exactly what I attempted on my first go round... and it got weird fast 🤣 

Here's the initial PR: https://github.com/jstime/jstime/pull/54
The repo to roll up the built-in: https://github.com/jstime/jstime-url

**Reply by MylesBorins** [reply]
This is the rollup config that was used to strip out some of the use of Node.js builtins that exist inside of @whatwgURL, since we do need to bundle all the dependencies. I also did some work in there to capture all the licenses of all the dependencies so we could vendor it properly with attribution.

https://github.com/jstime/jstime-url/blob/main/rollup.config.mjs#L30
